lib/semantics/tools.cc

```diff
@@ -508,6 +509,139 @@ std::optional<parser::MessageFixedText> WhyNotModifiable(
   }
 }
 
+std::optional<parser::MessageFixedText> GetImageControlStmtMsg(
+    const parser::SyncAllStmt &) {
+  return "SYNC ALL is not allowed in this context"_err_en_US;
+}
+
+std::optional<parser::MessageFixedText> GetImageControlStmtMsg(
+    const parser::SyncImagesStmt &) {
+  return "SYNC IMAGES is not allowed in this context"_err_en_US;
+}
+
+std::optional<parser::MessageFixedText> GetImageControlStmtMsg(
+    const parser::SyncMemoryStmt &) {
+  return "SYNC MEMORY is not allowed in this context"_err_en_US;
+}
+
+std::optional<parser::MessageFixedText> GetImageControlStmtMsg(
+    const parser::SyncTeamStmt &) {
+  return "SYNC TEAM is not allowed in this context"_err_en_US;
+}
+
+bool IsCoarray(const parser::AllocateObject &allocateObject) {
+  const parser::Name &name{GetLastName(allocateObject)};
+  if (name.symbol && IsCoarray(*name.symbol)) {
+    return true;
+  } else {
+    return false;
+  }
```

Comment:
This can be written as `return name.symbol && IsCoarray(*name.symbol);`

Reply:
Thanks.  I'll make that change.